### DISCUSSION TITLE
Why projects built with manifest mode uses caches different than system wide packages

### DISCUSSION THREAD

**Discussion by cppdev123** [microsoft/vcpkg#26912] [Q&A] [answered]
When installing a package via `vcpkg install` the package is cached inside the cache directory and this is fine. When installing this package as a part of building a project with manifest vcpkg will not pick the one in the previously cached one but will built it again and put it in the cache. The next time this package is installed in another project with manifest mode it is picked from the cache without compiling. Does vcpkg keeps separate caches for manifests and system wide packages ? why ? this causes unnecessary recompilation

**Comment by Osyotr** [comment]
> Does vcpkg keeps separate caches for manifests and system wide packages

Local binary cache in `%localappdata%\vcpkg` is used for both classic and manifest modes.

> this causes unnecessary recompilation

Check `vcpkg_abi_info.txt` to see what really caused cache invalidation.

**Comment by cppdev123** [comment]
it is `vcpkg-cmake` and `vcpkg-cmake-config` dependencies that are different because the system wide uses ones built for triplet `x64-windows` although the packages are `x64-windows-static` but when I specify for cmake the host target as `x64-windows-static` the same as `VCPKG_TARGET_TRIPLET` then  `vcpkg-cmake` and `vcpkg-cmake-config` are built for triplet `x64-windows-static` and the result packages are not the same because of these two dependencies

**Reply by Osyotr** [reply]
If `vcpkg-cmake` and `vcpkg-cmake-config` are installed in classic mode, they might be outdated. Try `vcpkg upgrade`.

**Comment by dg0yt** [comment, marked as answer]
`vcpkg-cmake` and similar script ports are installed for the host triplet. Normally this is what you want: Use the same cached binary package for code which is executed on the host. Don't change the host triplet unless it is wrongly detected by vcpkg.

If you change the host triplet for either classic mode or manifest mode, but not both, this will result in different binary artifacts.

**Reply by cppdev123** [reply]
the host triplet should not affect the compiled package itself, right ? so if `vcpkg` uses `x64-windows` as the host triplet for a `x64-windows-static` package then there should be no problems ?
